src/common/components/paymentMethods/deletePaymentMethod/deletePaymentMethod.modal.component.js

```diff
@@ -0,0 +1,276 @@
+
+let componentName = 'deletePaymentMethodModal';
+
+class deletePaymentMethodModalController {
+
+  /* @ngInject */
+  constructor(profileService, $log) {
+    this.profileService = profileService;
+    this.$log = $log;
+    this.loading = false;
+    this.view = '';
+    this.filteredPaymentMethods = [];
+    this.confirmText = '';
+    this.submitted = false;
+    this.submissionError = {
+      error: ''
+    };
+  }
+
+  $onInit() {
+    this.setView();
+    this.getPaymentMethods();
+  }
+
+  changeView(goBack){
+    this.submissionError.error = '';
+    if(goBack) {
+      this.setView();
+      return;
+    }
+    switch(this.deleteOption) {
+      case '1'://move to an existing paymentMethod
+        this.view = 'confirm';
+        this.confirmText = 'withTransfer';
+      break;
+      case '2'://move to a new paymentMethod
+        this.view = 'addPaymentMethod';
+      break;
+      case '3'://confirm to delete
+        this.view = 'confirm';
+        this.confirmText = 'withOutTransfer';
+      break;
+    }
+  }
+
+  setView(){
+    this.view = this.resolve.paymentMethod._recurringgifts[0].donations.length ? 'manageDonations' : 'confirm';
+  }
+
+  getPaymentMethodName(newPaymentMethod){
+    let paymentMethod = newPaymentMethod ? find(this.filteredPaymentMethods,(item)=>{
+      return item.self.uri == this.selectedPaymentMethod;
+    }) : this.resolve.paymentMethod;
+    return paymentMethod['card-type'] || paymentMethod['bank-name'];
+  }
+
+  getPaymentMethodLastFourDigits(newPaymentMethod){
+    let paymentMethod = newPaymentMethod ? find(this.filteredPaymentMethods,(item)=>{
+      return item.self.uri == this.selectedPaymentMethod;
+    }) : this.resolve.paymentMethod;
+    return paymentMethod['display-account-number'] || paymentMethod['card-number'];
+  }
+
+  getPaymentMethodOptionLabel(paymentMethod){
+    return (paymentMethod['bank-name'] || paymentMethod['card-type']) + ' ending in ****' + (paymentMethod['display-account-number'] || paymentMethod['card-number']);
+  }
+
+  getIcon(){
+    return this.resolve.paymentMethod['card-type'] ? 'cc-'+this.resolve.paymentMethod['card-type'].toLowerCase() : 'bank';
+  }
+
+  getRecurrence(gift){
+    var date = new Date(gift['next-draw-date']['display-value']),
+        month = date.getMonth()*1,
+        monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
+        text='';
+
+    switch(gift.rate.recurrence.interval){
+      case 'Quaterly':
+        var months = [];
+
+        for(let i=0;i<4;i++){
+          months.push(month);
+          month+=3;
+          month = month > 11 ? month - 12 : month;
+        }
+        months.sort((a,b) => {
+          return a-b;
+        });
+        text = `On the ${gift['recurring-day-of-month']}th day of each ${monthNames[months[0]]}, ${monthNames[months[1]]}, ${monthNames[months[2]]} and ${monthNames[months[3]]}`;
```

Comment:
I might try to do more of this in the template... You could use ng-switch there. Idk. As long as you can make this work with angular-gettext, I don't care where you do this logic though... I'm just more familiar with doing it in the template. I think you've got a few places where you are constructing strings in JS. Just make sure they'll be translated correctly. Angular filters can handle some of the localization stuff which might be nice in the future.